includes/abstracts/llms-abstract-session-data.php

```diff
@@ -110,15 +110,22 @@ public function get( $key, $default = false ) {
 	 *
 	 * @param string $key   The key of the session variable.
 	 * @param mixed  $value The value of the session variable.
-	 * @return void
+	 * @return mixed
 	 */
 	public function set( $key, $value ) {
 
-		if ( $value !== $this->get( $key ) ) {
+		/**
+		 * Using `isset()` allows us to explicitly save a value of `false`
+		 * since the `get()` method will return the default value `false` making it look
+		 * as if the value hasn't changed (when it actually has).
+		 */
+		if ( ! isset( $this->$key ) || $value !== $this->get( $key ) ) {
 			$this->data[ sanitize_key( $key ) ] = maybe_serialize( $value );
 			$this->is_clean                     = false;
 		}
 
+		return $this->get( $key );
```

Comment:
@eri-trabiccolo 

Setters really shouldn't return anything according to PHP `__set()`, for example.

WP Setters (like `update_post_meta()` returns the row ID or false.

I think that in general I don't know...

This function I added the return back because the original function that's being updated/replaced returned the value so I wanted to keep it that way for backwards compatibility.

A rule sounds nice however no matter what rule we create a lot of our functions are going to be in violation of that rule...

Reply:
Yeah I see, anyway I'm going to consider not returning anything as general rule, exceptions are exceptions. :)